spec/js/views/app_view_spec.js

```diff
@@ -11,22 +11,33 @@ define([
     }
   });
 
-  describe('render', function () {
-    it ('will get rendered', function () {
-      subject().render();
-      expect(subject().$el.text()).toEqual(jasmine.stringMatching(/The current date is:/));
+  describe('intial state', function(){
+    it('will be empty', function(){
+        subject().render();
+        expect(subject().collection.length).toBe(0);
     })
   });
+  describe('adding an items on click event', function(){
+    var spy;
+    beforeEach(function(){
+      spy = spyOn(subject(),'addItems');
+      subject().addItems('adga');
+    });
 
-  describe('currentDate', function () {
-    beforeEach(function () {
-      subject = function () {
-        return instance.currentDate();
-      }
+    it('should call click event handler',function(){
+      expect(spy).toHaveBeenCalled();
     });
 
-    it('returns a date', function () {
-      expect(subject() instanceof Date).toBe(true);
+  });
+
+  describe('adding an items on key event enter', function(){
+    var spy;
+    beforeEach(function(){
+     spy = spyOn(subject(),'keyAction');
+     subject().keyAction('adga');
+   });
+   it('should call key event handler',function(){
+      expect(spy).toHaveBeenCalled();
```

Comment:
You are stubbing a method, calling the same method and then verifying that the spy was called (which always is true, because you call it). This is not a test for your functionality, it actually tests if the spying framework works